### PR TITLE
[TEST, don't merge] windows build with test

### PR DESCRIPTION
**Pull request by mingbowan**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,5483 +13,71 @@
...
-            sudo chmod +x /usr/local/bin/sccache
-            export SCCACHE_BUCKET=ossci-compiler-cache-circleci-v2
-
-            # This IAM user allows write access to S3 bucket for sccache
+            set -ex
             set +x
-            export AWS_ACCESS_KEY_ID=${CIRCLECI_AWS_ACCESS_KEY_FOR_SCCACHE_S3_BUCKET_V4}
-            export AWS_SECRET_ACCESS_KEY=${CIRCLECI_AWS_SECRET_KEY_FOR_SCCACHE_S3_BUCKET_V4}
+            export AWS_ACCESS_KEY_ID=${MINGBO_AWS_KEY_ID}
+            export AWS_SECRET_ACCESS_KEY=${MINGBO_AWS_SECRET_KEY}
```
Comment (ezyang) [inline .circleci/config.yml]: To be fixed before merging right? :>

**Comment by ezyang** [comment]
Pleasingly simple!

File: .circleci/config.yml [hunk 2]
```diff
@@ -13,5483 +13,71 @@
...
-            # copy with -a to preserve relative structure (e.g., symlinks), and be recursive
-            cp -a ~/workspace/project/. ~/project
-
-            chmod a+x .jenkins/pytorch/macos-test.sh
-            unbuffer .jenkins/pytorch/macos-test.sh 2>&1 | ts
-
-  pytorch_macos_10_13_cuda9_2_cudnn7_py3_build:
+            choco install cmake --installargs 'ADD_CMAKE_TO_PATH=User'
+            choco install -y 7zip curl awscli
+            choco upgrade -y visualstudio2019community
```
Comment (peterjc123) [inline .circleci/config.yml]: Wow, do we have to do this every time before build?

File: .circleci/config.yml [hunk 3]
```diff
@@ -13,5483 +13,71 @@
...
+            pushd ..
+            curl -O https://ossci-windows.s3.amazonaws.com/cudnn-10.1-windows10-x64-v7.6.4.38.zip
+            7z x cudnn-10.1-windows10-x64-v7.6.4.38.zip -ocudnn
+            cp -r  cudnn/cuda/* "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/"
```
Comment (peterjc123) [inline .circleci/config.yml]: No cudnn pre-installed? OMG.

File: .circleci/config.yml [hunk 4]
```diff
@@ -13,5483 +13,71 @@
...
-          time docker push ${DEBUG_COMMIT_DOCKER_IMAGE}
-
-  pytorch_macos_10_13_py3_build:
+  pytorch_windows_build:
+    executor: windows-gpu-prototype
     environment:
-      BUILD_ENVIRONMENT: pytorch-macos-10.13-py3-build
-    macos:
-      xcode: "9.0"
+      BUILD_ENVIRONMENT: "pytorch-win-ws2016-cuda9-cudnn7-py3"
```
Comment (peterjc123) [inline .circleci/config.yml]: It is using cuda 10.1. Probably we need to change the name here.

File: .circleci/config.yml [hunk 5]
```diff
@@ -13,5483 +13,71 @@
...
+            cp -r  cudnn/cuda/* "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.1/"
+            popd
+            .jenkins/pytorch/win-build.sh
+  pytorch_windows_test:
+    executor: windows-gpu-prototype
     environment:
-      BUILD_ENVIRONMENT: pytorch-macos-10.13-cuda9.2-cudnn7-py3-build
-    macos:
-      xcode: "9.0"
+      BUILD_ENVIRONMENT: "pytorch-win-ws2016-cuda9-cudnn7-py3"
```
Comment (peterjc123) [inline .circleci/config.yml]: rename to cuda10.1 here

**Review by peterjc123 — APPROVED** [review]
Thanks for working on this.

File: .circleci/config.yml (hunk 3 above)
Comment (ezyang) [reply .circleci/config.yml]: Let's follow up with CircleCI on this. @mingbowan do you know how to get in touch with them?

File: .circleci/config.yml (hunk 1 above)
Comment (mingbowan) [reply .circleci/config.yml]: yes, current key pair has no rights to upload image to s3

File: .circleci/config.yml (hunk 2 above)
Comment (mingbowan) [reply .circleci/config.yml]: the visualstudio2019community upgrade is a workaround, once circleCI upgrade their base image, we don't have to. But for others, we need to install unless CircleCI include them in the base image

File: .circleci/config.yml (hunk 3 above)
Comment (mingbowan) [reply .circleci/config.yml]: I can ask them to install cudnn

File: .circleci/config.yml (hunk 4 above)
Comment (mingbowan) [reply .circleci/config.yml]: yes, will change on official PR.

File: .circleci/config.yml (hunk 2 above)
Comment (mingbowan) [reply .circleci/config.yml]: oops, didn't realized you (@peterjc123 ) approved the change. This one is purely for test, I removed all other builds. I'll send new PR soon

**Comment by peterjc123** [comment]
I'm okay with the changes. Just waiting for your final PR.

**Pull request closed without merging**